### DISCUSSION TITLE
I would like to share the result of my work for inspiration

### DISCUSSION THREAD

**Discussion by leiweibau** [netalertx/NetAlertX#159] [General]
Hello,

even though our forks have grown a bit apart, I'm still following your work, of course, and I'm also inspired by one or the other idea. :)

I'm working on an idea right now and want you to get something out of it too. 
I have developed a first Python prototype based on another fork, with which I plan to monitor webservices (in the own network or internet, if desired) and get notified if necessary.
e.g. the server is reachable, but the webserver has crashed.

Maybe you are interested in this, if not, it's no problem. 

It is currently a stand alone script. It is only the backend part at the moment. I am still far away from a release, but the prototype works for now.
[monitor.zip](https://github.com/jokob-sk/Pi.Alert/files/10557947/monitor.zip)

**Comment by jokob-sk** [comment]
Hey!
Great idea! 
I really appreciate it, I will have a look later this week /next one probably after this release.

What would be the advantage / added value of this over existing projects such as changedetection, update kuma etc?
Ps. Ido closely watch your fork too :) I like the webgui notification, still thinking about if to add it as well 😁
Thanks, 
J

**Comment by leiweibau** [comment]
Uptime-Kuma is much more complex than my implementation. I don't want to compete with that.

Since my fork is not based on Docker, I wanted to find a simple solution for myself that also does not require Docker. And since you're already doing advanced scans, I thought you might be interested.

Whether I extend the whole thing to other services later is up in the air.

**Comment by jokob-sk** [comment]
Makes sense. You should come over to the docker-dark-side 😁 I'll have a look at the script later. Thanks again for sharing 🙏

**Comment by jokob-sk** [comment]
Implemented as a custom plugin included in this release: https://github.com/jokob-sk/Pi.Alert/releases/tag/v23.03.11

**Comment by leiweibau** [comment]
That must have been the most complex modification to your fork so far, right? Great work.

**Comment by jokob-sk** [comment]
Yes, one of the most substantial ones and I think there will be a few bugs I missed, so anxious to see the reported issues lol
I hope this change gives me more flexibility to extend the functionality and iterate more quicker, let's see - I hope it doesn't cause me more headaches 😅

**Comment by leiweibau** [comment]
I received a feature request to detect Roque DHCP servers. Except for the notification, I already have it. Since you already do nmap scans yourself, I don't know at the moment if you can delete it by simple queries. If not, I can give you the code snippets.

**Reply by jokob-sk** [reply]
Oh, that would be amazing. I don't know how complex it would be to add it, but I'd have a look 👀

**Comment by leiweibau** [comment]
Unfortunately nmap needs sudo rights for this, but arp-scan does too

[code.txt](https://github.com/jokob-sk/Pi.Alert/files/10949482/code.txt)

**Reply by jokob-sk** [reply]
Thanks, implemented as a plugin: 
![image](https://user-images.githubusercontent.com/96159884/224533372-69c37728-aca2-4669-9e74-3cae86b30eb2.png)
